### ISSUE TITLE
RCS1176 shouldn't flag code inside of `fixed (...)`

### ISSUE THREAD

**Issue by xt0rted** [dotnet/roslynator#1290]
**Product and Version Used**: 4.6.4

**Steps to Reproduce**:

Enable `RCS1176` and set it's severity to `error`:

```.editorconfig
# RCS1176: Use 'var' instead of explicit type (when the type is not obvious)
dotnet_diagnostic.RCS1176.severity = error
```

Use the following code:

```csharp
var stringSpan = scriptName.AsSpan();

fixed (char* ptr = &MemoryMarshal.GetReference(stringSpan))
{
  ...
}
```

**Actual Behavior**:

1. `char*` will be flagged saying it should be replaced with `var`
2. Run the code fix to switch `char*` to `var`
3. Now `ptr = &MemoryMarshal.GetReference(scriptNameSpan)` will be flagged for [`CS0821`](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0821) and won't compile

**Expected Behavior**:

Ignore explicit types inside of `fixed (...)` since they can't be switched to use `var`.
